### DISCUSSION TITLE
Mapping a keyboard shortcut to clear terminal and redraw current line

### DISCUSSION THREAD

**Discussion by salmankhilji** [kovidgoyal/kitty#3654] [Ideas]
As suggested within the configuration docs, I have the following in my `kitty.conf` file:

```
map ctrl+l combine : clear_terminal scroll active : send_text normal,application \x0c
```

When there is no command running within bash (i.e., when it is showing the prompt), this works great: I can _clear_ the terminal _and_ force bash to redraw the prompt using a _single_ keyboard shortcut.

> Without the `combine` feature, I'd have to use two keyboard shortcuts: one to reset the terminal and another to redraw the current line.  And, of course, the `send_text` part is needed because bash otherwise would not redraw the prompt after the terminal is reset.

The issue is that when I have active long running command within bash, say `adb logcat`, then this keyboard shortcut produces an unwanted `^L` character to appear on the screen.  (This can, of course, be deleted using `BACKSPACE` key.)  The current workaround is again to map two keyboard shortcuts:  one to press when bash is not running a program and another to press when bash is running a long running program.  This is not ideal.

Is there anyway we can get kitty to execute the `send_text` command conditionally?  Perhaps it could be another cousin of `send_text` that internally would only execute if the underlying shell is _not_ running a command, like this:

```
map ctrl+l combine : clear_terminal scroll active : send_text_if_idle normal,application \x0c
```

However, I am not sure if it is even possible for kitty to detect whether a command is currently running within the shell or not.  Looking at the `confirm_os_window_close` option, it appears that kitty _may_ be able to detect whether the underlying shell is busy running another command or not.

**Comment by kovidgoyal** [comment]
It's not possible in general. Think of running a shell over ssh. Without co-operation from the shell there is no robust way to do this.

**Reply by salmankhilji** [reply]
> It's not possible in general. Think of running a shell over ssh. Without co-operation from the shell there is no robust way to do this.

Resurrecting this issue, now that we have shell integration, how hard would it be to implement a new action in kitty.conf to send text _only_ if the current program is a shell _and_ its sitting idle at a prompt?

I am currently using:

```
map ctrl+l clear_terminal to_cursor active
```

and this _mostly_ works.  However, it would be perfect if I could do:

```
map ctrl+l combine : clear_terminal to_cursor active : send_text_if_at_prompt normal,application \x0c
```

The new action, `send_text_if_at_prompt`, would be ignored if the shell was busy running some command.
